### PR TITLE
Added Dockerfile to build and run console Metrics viewer

### PR DESCRIPTION
**Pull request by brennangaunce**

### CODE REVIEW

**Comment by brennangaunce** [comment]
Irina - can you please review and update as needed?

File: src/main/docker/go.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+export GATEWAY_OPTS=-Dorg.kaazing.gateway.management.AGRONA_ENABLED=true
```
Comment (irina-mitrea-luxoft) [inline src/main/docker/go.sh:2]: We could use "ENV GATEWAY_OPTS -Dorg.kaazing.gateway.management.AGRONA_ENABLED=true" within the Dockerfile for this purpose.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,7 @@
+FROM kaazing/gateway:latest
```
Comment (irina-mitrea-luxoft) [inline Dockerfile:1]: The docker image (gateway version 5.0.1.21) on:
https://registry.hub.docker.com/u/kaazing/gateway/dockerfile/ 
seems to not have Agrona included yet.

Other available gateway versions are listed here: 
https://oss.sonatype.org/content/repositories/releases/org/kaazing/gateway.distribution/

Gateway version 5.0.1.27 seems to have Agrona available. 

Can we make a switch to this version?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# metrics
+Metrics gathering and monitoring for Kaazing Gateway
+
+```
+git clone https://github.com/kaazing/gateway.git
+pushd gateway
+mvn clean install -Pdocker
```
Comment (irina-mitrea-luxoft) [inline README.md:7]: Would a gateway clone be needed here? Shouldn't it be sufficient to have the metrics repo built and then running the Dockerfile?

File: Dockerfile (hunk 2 above)
Comment (irina-mitrea-luxoft) [inline Dockerfile:1]: Where should the Dockerfiles be placed?

We already have a Dockerfile/docker-compose located wthin tools.docker which is responsible with starting up the gateway and the metrics publisher and publishing the data to StatsD/Graphite, in the following forks:
- https://github.com/alingherasim/tools.docker/tree/develop/metrics (forked from kaazing/tools.docker)
- https://github.com/irina-mitrea-luxoft/tools.docker/tree/develop/metrics (forked from alingherasim/tools.docker)

My question is posted due to the fact that the metrics publisher Dockerfile is placed under tools.docker (not yet merged into the main repo though), while the metrics viewer Dockerfile in this pull request is placed under the metrics project. Should the metrics publisher Dockerfile also be placed under the metrics project for consistency?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+RUN mkdir /kaazing-metrics
+ADD viewer/target/metrics.viewer-develop-SNAPSHOT-jar-with-dependencies.jar /kaazing-metrics/
```
Comment (irina-mitrea-luxoft) [inline Dockerfile:4]: A build would need to be triggered prior to obtaining the metrics viewer jar.

**Comment by brennangaunce** [comment]
Not yet ready to merge.  Great comments Irina!

**Comment by brennangaunce** [comment]
Let's file a story for this, and consider for next sprint.  It looks like we have a few more items to cover - we should not need to build the metrics viewer within the docker image.  We should use a released version.

**Pull request closed without merging**